chunk/chunk_store.go

```diff
@@ -324,13 +329,13 @@ func (c *Store) lookupMatchers(ctx context.Context, userID string, from, through
 				incomingErrors <- err
 				return
 			}
-			incoming, err := c.lookupEntries(ctx, entries, matcher)
+			incoming, err := c.lookupEntries(ctx, entries, &matcher)
 			if err != nil {
 				incomingErrors <- err
 			} else {
 				incomingChunkSets <- incoming
 			}
-		}(matcher)
+		}(*matcher)
```

Comment:
Given that this turns out not to matter, and makes the code harder to follow (IMO), should it be reverted?